### PR TITLE
issue #2522 - Mark everything related to the old query builder deprecated

### PR DESCRIPTION
**Pull request by lmsurpre**
I also removed the previously deprecated fhir-persistence-proxy module.

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

**Review by prb112 — APPROVED** [review]

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java [hunk 1]
```diff
@@ -223,6 +226,7 @@ public SqlQueryData buildQuery(Class<?> resourceType, FHIRSearchContext searchCo
      *         query segments.
      * @throws Exception
      */
+    @Deprecated
```
[the remark this answers is not in the document]
Comment (lmsurpre) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java:229]: it was just a best-effort at finding everything that was only used by the old query builder...would love some help identifying additional methods if you have suggestions

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/api/ResourceDAO.java [hunk 2]
```diff
@@ -84,6 +84,7 @@ int historyCount(String resourceType, String logicalId, Timestamp fromDateTime)
      * @throws FHIRPersistenceDataAccessException
      * @throws FHIRPersistenceDBConnectException
      */
+    @Deprecated
```
[the remark this answers is not in the document]
Comment (lmsurpre) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/api/ResourceDAO.java:87]: great, I'll mark those as deprecated too

[the remark this answers is not in the document]
Comment (lmsurpre) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/api/ResourceDAO.java:87]: done

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (lmsurpre) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java:229]: I don't think its too bad if we miss some private methods somewhere...the goal was to mark everything that will get removed but I think we've done enough to get the point across :-)

[the remark this answers is not in the document]
Comment (michaelwschroeder) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java:229]: I'm good with what you have. I think marking the public ones is sufficient.

**Review by michaelwschroeder — APPROVED** [review]
LGTM

**Pull request merged**